### PR TITLE
Add support for the logger private buffer pool.

### PR DESCRIPTION
**Pull request by edoger**
Add a private buffer pool for the logger. By default, the default global buffer pool will be used.

### CODE REVIEW

**Comment by edoger** [comment]
link #1247

File: entry.go [hunk 1]
```diff
@@ -260,6 +261,13 @@ func (entry *Entry) log(level Level, msg string) {
 	}
 }
 
+func (entry *Entry) getBufferPool() (pool BufferPool) {
+	if entry.Logger.BufferPool != nil {
```
Comment (dgsb) [inline entry.go:265]: Can we have that protected by the logger mutex ?

**Review by dgsb — CHANGES_REQUESTED** [review]
Thanks for your contribution @edoger

I just have a small comment regarding the missing lock when getting the buffer pool from the logger.

File: entry.go (hunk 1 above)
Comment (edoger) [reply entry.go:265]: The only place to call this method is already protected by a mutex.

Comment (dgsb) [reply entry.go:265]: ok, I missed that 👍

**Review by dgsb — APPROVED** [review]

**Pull request merged**